### PR TITLE
Update HelloAndroidCamera2 to use Halide::Buffer

### PR DESCRIPTION
**Pull request by abadams**
@jiawen PTAL when you get a chance

The goal here is to make it easier to migrate  from buffer_t to halide_buffer_t by minimizing direct use of buffer_t

### CODE REVIEW

**Comment by abadams** [comment]
Tested on pixel and 6p

File: apps/HelloAndroidCamera2/jni/Android.mk [hunk 1]
```diff
@@ -6,15 +6,13 @@ LOCAL_MODULE    := HelloAndroidCamera2
...
     HalideFilters.cpp \
     LockedSurface.cpp \
-    YuvBufferT.cpp \
-    YuvFunctions.cpp
+    YuvBufferT.cpp
 LOCAL_LDFLAGS   := -L$(LOCAL_PATH)/../jni
 LOCAL_LDLIBS    := -lm -llog -landroid -latomic
-LOCAL_LDLIBS    += $(LOCAL_PATH)/../bin/$(TARGET_ARCH_ABI)/deinterleave.a
-LOCAL_LDLIBS    += $(LOCAL_PATH)/../bin/$(TARGET_ARCH_ABI)/edge_detect.a
+LOCAL_LDLIBS += $(LOCAL_PATH)/../bin/$(TARGET_ARCH_ABI)/deinterleave.a
```
Comment (jiawen) [inline apps/HelloAndroidCamera2/jni/Android.mk]: Uber-nit: make spacing consistent. Just make it all one space?

File: apps/HelloAndroidCamera2/jni/HalideFilters.cpp [hunk 2]
```diff
@@ -35,6 +34,29 @@ extern "C" void halide_print(void *, const char *msg) {
 
 extern "C" {
 
+bool assertEqualExtents(YuvBufferT *src, YuvBufferT *dst) {
```
Comment (jiawen) [inline apps/HelloAndroidCamera2/jni/HalideFilters.cpp]: Rename back to equalExtents(). assert is weird. I would also prefer const ref arguments over pointer.

(Don't feel that strongly) Having the LOGE inside is also a bit weird. I see that having the message duplicated outside is also annoying. Maybe make a separate function?

**Comment by steven-johnson** [comment]
LGTM

**Pull request merged**